arch/x86/boot/multiboot.c

```diff
@@ -182,6 +182,13 @@ void init_multiboot(unsigned long *addr, const char **cmdline) {
     }
 }
 
+bool mbi_reserved_range(addr_range_t *reserved_range) {
+    reserved_range->start = multiboot2_hdr;
+    reserved_range->end = multiboot2_hdr + multiboot2_hdr_size;
```

Comment:
since multiboot2_hdr is a pointer (void, but still), it makes sense to cast it explicitly